torch/nn/attention/_flex_attention.py

```diff
@@ -65,59 +111,57 @@ def _no_mask(
 
 
 class BlockMask:
-    full_kv_num_blocks: torch.Tensor
-    full_kv_indices: torch.Tensor
-    full_q_num_blocks: torch.Tensor
-    full_q_indices: torch.Tensor
-    partial_kv_num_blocks: torch.Tensor
-    partial_kv_indices: torch.Tensor
-    partial_q_num_blocks: torch.Tensor
-    partial_q_indices: torch.Tensor
+    kv_num_blocks: torch.Tensor
+    kv_indices: torch.Tensor
+    q_num_blocks: torch.Tensor
+    q_indices: torch.Tensor
+    full_kv_num_blocks: Optional[torch.Tensor]
+    full_kv_indices: Optional[torch.Tensor]
+    full_q_num_blocks: Optional[torch.Tensor]
+    full_q_indices: Optional[torch.Tensor]
     KV_BLOCK_SIZE: int
```

Comment:
I renamed the "full vs partial" blocks.

# Before:
 **no mask fn**
full_kv_num_blocks = all the blocks computed
partial_kv_num_blocks = placeholder empty value with all zeros
** with mask fn**
full_kv_num_blocks = only the blocks that are completely full
partial_kv_num_blocks = blocks that aren't full and aren't empty.

# After:
 **no mask fn**
kv_num_blocks = all the blocks computed
full_kv_num_blocks = None
** with mask fn**
kv_num_blocks = the blocks that are partially computed
full_kv_num_blocks = only the blocks that we know are completely full

This makes more sense to me. Basically, `kv_num_blocks` are blocks that we don't know anything special about. So we always need to apply both the `score_mod` and the `mask_mod`. If passed a `mask_mod` to BlockMask and given `full_kv_num_blocks`, then we know have some blocks that we know are "special" and don't have masking applied. So we only need to apply `score_mod`.